tembo-pgmq-python/tembo_pgmq_python/async_queue.py

```diff
@@ -198,6 +219,29 @@ async def _send_batch_internal(self, queue, messages, delay, conn):
         self.logger.debug(f"Batch messages sent with msg_ids={msg_ids}")
         return msg_ids
 
+    @transaction
+    async def send_batch_at(self, queue: str, messages: List[dict], delay: str, conn=None) -> List[int]:
+        """Send a batch of messages to a queue with timestamp."""
+        self.logger.debug(f"send_batch_at called with queue='{queue}', messages={messages}, delay={delay}, conn={conn}")
+        if conn is None:
+            async with self.pool.acquire() as conn:
+                return await self._send_batch_at_internal(queue, messages, delay, conn)
+        else:
+            return await self._send_batch_at_internal(queue, messages, delay, conn)
+
+    async def _send_batch_at_internal(self, queue, messages, delay, conn):
+        self.logger.debug(f"Sending batch of messages to queue '{queue}' with delay={delay}")
+        jsonb_array = [dumps(message).decode("utf-8") for message in messages]
+        result = await conn.fetch(
+            "SELECT * FROM pgmq.send_batch_at($1, $2::jsonb[], $3);",
+            queue,
+            jsonb_array,
+            delay,
+        )
+        msg_ids = [message[0] for message in result]
+        self.logger.debug(f"Batch messages sent with msg_ids={msg_ids}")
+        return msg_ids
```

Comment:
@tavallaie - what are you opinion on function naming in the python client? I'm more in favor of overloading the function names in Postgres, but in the python client I am not sure.

Reply:
In python, function overloading isn't supported. It's easier to maintain code with separate, descriptive function names. so those names are good in my opinion.

Reply:
While overloading in python is not supported, parameters can accept multiple types. For example, `send(... delay: int, vt: Option[Datetime] = None)` is valid, right?

I'm ok just going with `send_batch()` in the Python client though.

Reply:
or even something this would be valid:

```python
async def _send_batch_at_internal(self, queue, messages, vt: Union[int, Datetime], conn):
   if isinstance(vt, int) ....
   else ...
```

Reply:
is it possible if someone use both together? if not we can add some exception to prevent that.